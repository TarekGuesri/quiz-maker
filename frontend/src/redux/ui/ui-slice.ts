import { createSlice } from "@reduxjs/toolkit";
import { UIState } from "src/types";

const initialState: UIState = {
  darkMode: localStorage.getItem("darkMode") === "on",
};

export const uiSlice = createSlice({
  name: "ui",
  initialState,
  // The `reducers` field lets us define reducers and generate associated actions
  reducers: {
    switchDarkMode: (state) => {
      state.darkMode = !state.darkMode;

      localStorage.setItem("darkMode", state.darkMode ? "on" : "off");
    },
  },
});

export const { switchDarkMode } = uiSlice.actions;

export default uiSlice.reducer;
