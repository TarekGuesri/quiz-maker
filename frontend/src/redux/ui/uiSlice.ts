import { createSlice } from '@reduxjs/toolkit';

export interface UIState {
  darkMode: boolean;
}

const initialState: UIState = {
  darkMode: true,
};

export const uiSlice = createSlice({
  name: 'ui',
  initialState,
  // The `reducers` field lets us define reducers and generate associated actions
  reducers: {},
});

export default uiSlice.reducer;
