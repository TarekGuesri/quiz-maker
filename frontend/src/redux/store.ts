import { Action, ThunkAction, configureStore } from "@reduxjs/toolkit";

import ui from "./ui/ui-slice";

export const store = configureStore({
  reducer: {
    ui,
  },
  devTools: true,
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
