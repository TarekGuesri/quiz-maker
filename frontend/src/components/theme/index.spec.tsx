import React from "react";
import { cleanup, render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "src/redux/store";
import { Theme } from ".";
import { switchDarkMode } from "src/redux/ui/ui-slice";

describe("src/components/theme", () => {
  afterEach(() => cleanup());

  test("Render Theme component when darkMode is false", () => {
    const renderedComponent = render(
      <Provider store={store}>
        <Theme>Test </Theme>
      </Provider>,
    );

    expect(renderedComponent.container).toMatchSnapshot();
  });

  test("Render Theme component when darkMode is true", () => {
    store.dispatch(switchDarkMode());

    const renderedComponent = render(
      <Provider store={store}>
        <Theme>Test </Theme>
      </Provider>,
    );

    expect(renderedComponent.container).toMatchSnapshot();
  });
});
