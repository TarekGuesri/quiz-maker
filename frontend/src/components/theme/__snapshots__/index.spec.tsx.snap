// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/theme Render Theme component when darkMode is false 1`] = `
<div>
  Test 
</div>
`;

exports[`src/components/theme Render Theme component when darkMode is true 1`] = `
<div>
  Test 
</div>
`;
